<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.24.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Notes of PICRUST - Jie Li (She/Her)</title>
<meta name="description" content="1. Introduction Picrust是一款基于16S rRNA gene数据库对amplicon测序的群落进行功能预测的工具。目前16S rRNA gene有并且只有13_5和12_8两个版本。 picrust是基于已经计算好的数据库进行预测，数据库下载地址。 下载的数据库需要放到Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/目录下；因为picrust软件默认是访问这个目录来找数据库">


  <meta name="author" content="Jie Li">
  
  <meta property="article:author" content="Jie Li">
  


<meta property="og:type" content="article">
<meta property="og:locale" content="en_AU">
<meta property="og:site_name" content="Jie Li (She/Her)">
<meta property="og:title" content="Notes of PICRUST">
<meta property="og:url" content="http://localhost:3080/bioinformatics/amplicon/PICRUST/">


  <meta property="og:description" content="1. Introduction Picrust是一款基于16S rRNA gene数据库对amplicon测序的群落进行功能预测的工具。目前16S rRNA gene有并且只有13_5和12_8两个版本。 picrust是基于已经计算好的数据库进行预测，数据库下载地址。 下载的数据库需要放到Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/目录下；因为picrust软件默认是访问这个目录来找数据库">





  <meta name="twitter:site" content="@jlli6t">
  <meta name="twitter:title" content="Notes of PICRUST">
  <meta name="twitter:description" content="1. Introduction Picrust是一款基于16S rRNA gene数据库对amplicon测序的群落进行功能预测的工具。目前16S rRNA gene有并且只有13_5和12_8两个版本。 picrust是基于已经计算好的数据库进行预测，数据库下载地址。 下载的数据库需要放到Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/目录下；因为picrust软件默认是访问这个目录来找数据库">
  <meta name="twitter:url" content="http://localhost:3080/bioinformatics/amplicon/PICRUST/">

  
    <meta name="twitter:card" content="summary">
    
  

  



  <meta property="article:published_time" content="2018-04-20T00:00:00+10:00">



  <meta property="article:modified_time" content="2023-01-23T00:00:00+10:00">



  

  


<link rel="canonical" href="http://localhost:3080/bioinformatics/amplicon/PICRUST/">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": "jie-li-3201a57b",
      "url": "http://localhost:3080/",
      "sameAs": ["https://twitter.com/jlli6t","https://www.linkedin.com/in/jie-li-3201a57b/","https://github.com/"]
    
  }
</script>







<!-- end _includes/seo.html -->




<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="preload" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
<noscript><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css"></noscript>



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    

<div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        
        <a class="site-title" href="/">
          Jie Li (She/Her)
          <span class="site-subtitle"></span>
        </a>
        <ul class="visible-links"><li class="masthead__menu-item">
              <a href="/about/"><u>About</u></a>
            </li><li class="masthead__menu-item">
              <a href="/index.html"><u>Posts</u></a>
            </li><li class="masthead__menu-item">
              <a href="/projects/"><u>Projects</u></a>
            </li><li class="masthead__menu-item">
              <a href="/publications/"><u>Publications</u></a>
            </li><li class="masthead__menu-item">
              <a href="/media/"><u>Media</u></a>
            </li></ul>
        
        <button class="search__toggle" type="button">
          <span class="visually-hidden">Toggle search</span>
          <i class="fas fa-search"></i>
        </button>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      




  
    



<nav class="breadcrumbs">
  <ol itemscope itemtype="https://schema.org/BreadcrumbList">
    
    
    
      
        <li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
          <a href="/" itemprop="item"><span itemprop="name">Home</span></a>

          <meta itemprop="position" content="1" />
        </li>
        <span class="sep">/</span>
      
      
        
        <li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
          <a href="/categories/bioinformatics" itemprop="item"><span itemprop="name">Bioinformatics</span></a>
          <meta itemprop="position" content="2" />
        </li>
        <span class="sep">/</span>
      
    
      
      
        
        <li itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem">
          <a href="/categories/amplicon" itemprop="item"><span itemprop="name">Amplicon</span></a>
          <meta itemprop="position" content="3" />
        </li>
        <span class="sep">/</span>
      
    
      
      
        <li class="current">Notes of PICRUST</li>
      
    
  </ol>
</nav>

  


<div id="main" role="main">
  
  <div class="sidebar sticky">
  


<div itemscope itemtype="https://schema.org/Person" class="h-card">

  
    <div class="author__avatar">
      <a href="http://localhost:3080/">
        <img src="/assets/prof_pic2.jpg" alt="Jie Li" itemprop="image" class="u-photo">
      </a>
    </div>
  

  <div class="author__content">
    <h3 class="author__name p-name" itemprop="name">
      <a class="u-url" rel="me" href="http://localhost:3080/" itemprop="url">Jie Li</a>
    </h3>
    
      <div class="author__bio p-note" itemprop="description">
        

      </div>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      
        <li itemprop="homeLocation" itemscope itemtype="https://schema.org/Place">
          <i class="fas fa-fw fa-map-marker-alt" aria-hidden="true"></i> <span itemprop="name" class="p-locality">Earth</span>
        </li>
      

      
        
          
            <li><a href="https://scholar.google.com/citations?hl=zh-CN&user=s_Uga6sAAAAJ" rel="nofollow noopener noreferrer me" itemprop="sameAs"><i class="fab fa-google" aria-hidden="true"></i><span class="label">Google scholar</span></a></li>
          
        
          
            <li><a href="https://github.com/jlli6t" rel="nofollow noopener noreferrer me" itemprop="sameAs"><i class="fab fa-fw fa-github" aria-hidden="true"></i><span class="label">Github</span></a></li>
          
        
          
            <li><a href="https://www.linkedin.com/in/jie-li-3201a57b/" rel="nofollow noopener noreferrer me" itemprop="sameAs"><i class="fab fa-linkedin" aria-hidden="true"></i><span class="label">Linkedin</span></a></li>
          
        
          
            <li><a href="https://orcid.org/0000-0002-9623-9213" rel="nofollow noopener noreferrer me" itemprop="sameAs"><i class="fa-brands fa-orcid" aria-hidden="true"></i><span class="label">ORCID</span></a></li>
          
        
          
            <li><a href="https://twitter.com/jlli6t" rel="nofollow noopener noreferrer me" itemprop="sameAs"><i class="fab fa-fw fa-twitter-square" aria-hidden="true"></i><span class="label">Twitter</span></a></li>
          
        
      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs" rel="nofollow noopener noreferrer me">
      <i class="fas fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->
    </ul>
  </div>
</div>

  
  </div>



  <article class="page h-entry" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Notes of PICRUST">
    <meta itemprop="description" content="1. IntroductionPicrust是一款基于16S rRNA gene数据库对amplicon测序的群落进行功能预测的工具。目前16S rRNA gene有并且只有13_5和12_8两个版本。picrust是基于已经计算好的数据库进行预测，数据库下载地址。下载的数据库需要放到Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/目录下；因为picrust软件默认是访问这个目录来找数据库">
    <meta itemprop="datePublished" content="2018-04-20T00:00:00+10:00">
    <meta itemprop="dateModified" content="2023-01-23T00:00:00+10:00">

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title p-name" itemprop="headline">
            <a href="http://localhost:3080/bioinformatics/amplicon/PICRUST/" class="u-url" itemprop="url">Notes of PICRUST
</a>
          </h1>
          

  <p class="page__meta">
    
      
      <span class="page__meta-date">
        <i class="far fa-calendar-alt" aria-hidden="true"></i>
        
        <time datetime="2018-04-20T00:00:00+10:00">April 20, 2018</time>
      </span>
    

    <span class="page__meta-sep"></span>

    
      
      

      <span class="page__meta-readtime">
        <i class="far fa-clock" aria-hidden="true"></i>
        
          2 minute read
        
      </span>
    
  </p>


        </header>
      

      <section class="page__content e-content" itemprop="text">
        
          <aside class="sidebar__right ">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-heart"></i> Notes of PICRUST</h4></header>
              <ul class="toc__menu"><li><a href="#1-introduction">1. Introduction</a></li><li><a href="#2-installation">2. Installation</a></li><li><a href="#3-usage-tips">3. Usage tips</a><ul><li><a href="#31-way-1">3.1 way 1</a><ul><li><a href="#311-step-1">3.1.1 step 1</a></li><li><a href="#312-step-2">3.1.2 step 2</a></li></ul></li></ul></li><li><a href="#4-测试picrust-113">4. 测试PICRUST 1.1.3</a><ul><li><a href="#41-介绍">4.1 介绍</a></li><li><a href="#42-测试过程">4.2 测试过程</a><ul><li><a href="#421-生成biom文件">4.2.1 生成biom文件</a><ul><li><a href="#4211-方式1">4.2.1.1 方式1</a></li><li><a href="#4212-方式2">4.2.1.2 方式2</a></li></ul></li><li><a href="#422-对物种注释信息标准化">4.2.2 对物种注释信息标准化</a></li><li><a href="#423-功能预测">4.2.3 功能预测</a></li><li><a href="#424-对预测的功能进行map划分">4.2.4 对预测的功能进行map划分</a></li><li><a href="#425-是否可以用每一个通路基因在各个样品之间的值来做标准化">4.2.5 是否可以用每一个通路/基因在各个样品之间的值来做标准化。</a></li><li><a href="#426-待测试">4.2.6 待测试</a></li></ul></li></ul></li></ul>

            </nav>
          </aside>
        
        <h2 id="1-introduction">1. Introduction</h2>
<p><code class="language-plaintext highlighter-rouge">Picrust</code>是一款基于<code class="language-plaintext highlighter-rouge">16S rRNA gene</code>数据库对<code class="language-plaintext highlighter-rouge">amplicon</code>测序的群落进行功能预测的工具。目前<code class="language-plaintext highlighter-rouge">16S rRNA gene</code>有并且只有<code class="language-plaintext highlighter-rouge">13_5</code>和<code class="language-plaintext highlighter-rouge">12_8</code>两个版本。<br />
<code class="language-plaintext highlighter-rouge">picrust</code>是基于已经计算好的数据库进行预测，数据库<a href="http://picrust.github.io/picrust/picrust_precalculated_files.html">下载地址</a>。<br />
下载的数据库需要放到<code class="language-plaintext highlighter-rouge">Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/</code>目录下；因为<code class="language-plaintext highlighter-rouge">picrust</code>软件默认是访问这个目录来找数据库
<!--more--></p>

<h2 id="2-installation">2. Installation</h2>
<p>安装<code class="language-plaintext highlighter-rouge">picrust</code>, 这个中间没有遇到麻烦；注意<code class="language-plaintext highlighter-rouge">picrsut</code>是基于<code class="language-plaintext highlighter-rouge">python2.7</code>版本(用的<code class="language-plaintext highlighter-rouge">2.7.12</code>)， 其他版本的<code class="language-plaintext highlighter-rouge">python</code>会报错例如<code class="language-plaintext highlighter-rouge">python3.5.2</code>。</p>

<h2 id="3-usage-tips">3. Usage tips</h2>
<p>因为<code class="language-plaintext highlighter-rouge">picrust</code>是读入<code class="language-plaintext highlighter-rouge">biom</code>文件所以先要生成<code class="language-plaintext highlighter-rouge">biom</code>格式的文件，其中<code class="language-plaintext highlighter-rouge">biom</code>需要包含的信息有每个样品注释上的物种信息及对应的丰度，生成<code class="language-plaintext highlighter-rouge">biom</code>的格式方法有两种：</p>

<h3 id="31-way-1">3.1 way 1</h3>
<p>通过利用<code class="language-plaintext highlighter-rouge">qiime1</code>的命令先挑选<code class="language-plaintext highlighter-rouge">otu</code>比对的<code class="language-plaintext highlighter-rouge">ref</code>, 这里相当于重新比对分类一次，默认方法是<code class="language-plaintext highlighter-rouge">uclust</code>(其他还有<code class="language-plaintext highlighter-rouge">usearch_ref</code>， <code class="language-plaintext highlighter-rouge">sortmerna</code>), 所以挑出来的<code class="language-plaintext highlighter-rouge">otu</code>比<code class="language-plaintext highlighter-rouge">OTU_final.fasta</code>里的序列要少, 整体预测的通路的矩阵里的值会小一些，通路数量上可能也小一些。这里好像不支持<code class="language-plaintext highlighter-rouge">rdp</code>的结果，因为<code class="language-plaintext highlighter-rouge">rdp</code>基于贝叶斯的概率方法，最后没有给出具体比对的哪个<code class="language-plaintext highlighter-rouge">id</code>, 而是只给出一个分类<code class="language-plaintext highlighter-rouge">taxonomy</code>信息。</p>

<h4 id="311-step-1">3.1.1 step 1</h4>
<p>首先用<code class="language-plaintext highlighter-rouge">qiime1</code>里的<code class="language-plaintext highlighter-rouge">pick_closed_reference_otus.py</code>命令挑选<code class="language-plaintext highlighter-rouge">ref</code>：(可以使用<code class="language-plaintext highlighter-rouge">97</code>，也可以使用<code class="language-plaintext highlighter-rouge">90</code>，<code class="language-plaintext highlighter-rouge">99</code>或者其他的)</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">echo</span> <span class="s2">"pick_otus:enable_rev_strand_match True"</span><span class="o">&gt;&gt;</span>otu_picking_params_97.txt
<span class="nb">echo</span> <span class="s2">"pick_otus:similarity 0.97"</span><span class="o">&gt;&gt;</span>otu_picking_params_97.txt
pick_closed_reference_otus.py <span class="nt">-i</span> OTU_final.fasta <span class="nt">-o</span> <span class="nv">$PWD</span>/ucrC972/ <span class="nt">-p</span> <span class="nv">$PWD</span>/otu_picking_params_97.txt <span class="nt">-r</span> ./gg_13_5_otus/rep_set/97_otus.fasta <span class="nt">-t</span> ./gg_13_5_otus/taxonomy/97_otu_taxonomy.txt <span class="nt">-a</span> <span class="nt">-O</span> 8</code></pre></figure>

<h4 id="312-step-2">3.1.2 step 2</h4>
<p>使用命令<code class="language-plaintext highlighter-rouge">biom convert -i ucrC972/otu_table.biom -o ucrC972/otu_table.biom.tsv --to-tsv</code>
<code class="language-plaintext highlighter-rouge">tsv</code>文件如下图：主要是得到每个<code class="language-plaintext highlighter-rouge">otu</code>比对上的<code class="language-plaintext highlighter-rouge">ref</code>, 然后形成一个<code class="language-plaintext highlighter-rouge">0 1</code>矩阵
<img src="/assets/picbed/tools/picrust/picrust_biom2matrix.png" alt="" /></p>

<h2 id="4-测试picrust-113">4. 测试PICRUST 1.1.3</h2>
<h3 id="41-介绍">4.1 介绍</h3>
<p>根据<code class="language-plaintext highlighter-rouge">16S</code>的测序数据，对样本的现在通路进行预测。<code class="language-plaintext highlighter-rouge">picrust</code>只能基于<code class="language-plaintext highlighter-rouge">greengenes database</code>进行预测，并且只有<code class="language-plaintext highlighter-rouge">13_5</code>和<code class="language-plaintext highlighter-rouge">12_8</code>两个版本。<code class="language-plaintext highlighter-rouge">picrust</code>是基于已经计算好的数据库进行预测，数据库<a href="http://picrust.github.io/picrust/picrust_precalculated_files.html">下载地址</a></p>

<p>下载的数据库需要放到<code class="language-plaintext highlighter-rouge">Python-2.7.12/lib/python2.7/site-packages/PICRUSt-1.1.3-py2.7.egg/picrust/data/</code>目录下；因为<code class="language-plaintext highlighter-rouge">picrust</code>软件默认是访问这个目录来找数据库</p>

<h3 id="42-测试过程">4.2 测试过程</h3>
<h4 id="421-生成biom文件">4.2.1 生成biom文件</h4>
<p>因为<code class="language-plaintext highlighter-rouge">picrust</code>是读入<code class="language-plaintext highlighter-rouge">biom</code>文件，所以先要生成<code class="language-plaintext highlighter-rouge">biom</code>格式的文件。其中<code class="language-plaintext highlighter-rouge">biom</code>需要包含的信息有每个样品注释上的物种信息及对应的丰度，生成<code class="language-plaintext highlighter-rouge">biom</code>的格式方法有两种：</p>

<h5 id="4211-方式1">4.2.1.1 方式1</h5>
<p>通过利用<code class="language-plaintext highlighter-rouge">qiime1</code>的命令先挑选otu比对的<code class="language-plaintext highlighter-rouge">ref</code>, 这里相当于重新比对分类一次，默认方法是<code class="language-plaintext highlighter-rouge">uclust</code>(其他还有<code class="language-plaintext highlighter-rouge">usearch_ref</code>， <code class="language-plaintext highlighter-rouge">sortmerna</code>)。所以挑出来的<code class="language-plaintext highlighter-rouge">otu</code>比<code class="language-plaintext highlighter-rouge">OTU_final.fasta</code>里的序列要少, 整体预测的通路的矩阵里的值会小一些，通路数量上可能也小一些。这里好像不支持<code class="language-plaintext highlighter-rouge">rdp</code>的结果，因为<code class="language-plaintext highlighter-rouge">rdp</code>基于贝叶斯的概率方法，最后没有给出具体比对的哪个<code class="language-plaintext highlighter-rouge">id</code>, 而是只给出一个分类<code class="language-plaintext highlighter-rouge">taxonomy</code>信息。
    1. 首先用<code class="language-plaintext highlighter-rouge">qiime1</code>里的<code class="language-plaintext highlighter-rouge">pick_closed_reference_otus.py</code>命令挑选<code class="language-plaintext highlighter-rouge">ref</code>：(可以使用<code class="language-plaintext highlighter-rouge">97</code>，也可以使用<code class="language-plaintext highlighter-rouge">90</code>，<code class="language-plaintext highlighter-rouge">99</code>或者其他的)</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">echo</span> <span class="s2">"pick_otus:enable_rev_strand_match True"</span><span class="o">&gt;&gt;</span>otu_picking_params_97.txt
<span class="nb">echo</span> <span class="s2">"pick_otus:similarity 0.97"</span><span class="o">&gt;&gt;</span>otu_picking_params_97.txt
pick_closed_reference_otus.py <span class="nt">-i</span> OTU_final.fasta <span class="nt">-o</span> <span class="nv">$PWD</span>/ucrC972/ <span class="nt">-p</span> <span class="nv">$PWD</span>/otu_picking_params_97.txt <span class="nt">-r</span> ./gg_13_5_otus/rep_set/97_otus.fasta <span class="nt">-t</span> ./gg_13_5_otus/taxonomy/97_otu_taxonomy.txt <span class="nt">-a</span> <span class="nt">-O</span> 8</code></pre></figure>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>2. 将`biom`文件转换为`tsv`文件: 
</code></pre></div></div>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">biom convert <span class="nt">-i</span> ucrC972/otu_table.biom <span class="nt">-o</span> ucrC972/otu_table.biom.tsv <span class="nt">--to-tsv</span></code></pre></figure>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>`tsv`文件如下图：主要是得到每个`otu`比对上的`ref`, 然后形成一个`0 1`矩阵: ![](/assets/picbed/tools/picrust/picrust_biom2matrix.png)
3. 结合`OTU_shared_final.xls`的每个样品的每个`otu`的绝对丰度文件:
</code></pre></div></div>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">perl <span class="nt">-e</span> <span class="s1">'open(IN,"ucrC972/otu_table.biom.tsv");$t=&lt;IN&gt;;print"$t";%hash;$t=&lt;IN&gt;;chop($t);@t=split/\t/,$t;while(&lt;IN&gt;){chomp;@a=split;for($i=1;$i&lt;@a;$i++){if($a[$i]==1){$hash{$t[$i]}=$a[0];}}}close IN;$t=&lt;&gt;;chomp($t);print"#OTU ID$t\ttaxonomy\n";@t=split/\s+/,$t;%out;while(&lt;&gt;){chomp;@a=split;next if(not exists $hash{$a[0]});for($i=1;$i&lt;@a;$i++){$out{$hash{$a[0]}}{$t[$i]}+=$a[$i];}}foreach $k(sort keys %out){print"$k";for($i=1;$i&lt;@t;$i++){print"\t$out{$k}{$t[$i]}";}print"\n";}'</span> ../../OTU_shared_final.xls <span class="o">&gt;</span>otu_table.biom.txt</code></pre></figure>

<p>生成的<code class="language-plaintext highlighter-rouge">otu_table.biom.txt</code>文件如下格式：
<img src="/assets/picbed/tools/picrust/picrust_otu_table.png" alt="" />
    4. 将文件<code class="language-plaintext highlighter-rouge">convert</code>为<code class="language-plaintext highlighter-rouge">biom</code>格式:</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">biom convert <span class="nt">-i</span> otu_table.biom.txt <span class="nt">-o</span> otu_table.biom <span class="nt">--to-hdf5</span></code></pre></figure>

<h5 id="4212-方式2">4.2.1.2 方式2</h5>
<p>利用<code class="language-plaintext highlighter-rouge">mothur</code>软件的<code class="language-plaintext highlighter-rouge">make.biom</code>命令生成<code class="language-plaintext highlighter-rouge">biom</code>文件，直接将注释的结果转换成如上格式，需要用到<code class="language-plaintext highlighter-rouge">otu</code>的丰度值以及具体的<code class="language-plaintext highlighter-rouge">greengenes</code>数据库的注释信息。需要利用到注释的中间其他文件例如将<code class="language-plaintext highlighter-rouge">otu id</code>与<code class="language-plaintext highlighter-rouge">ref id</code>对应的文件。因为没有<code class="language-plaintext highlighter-rouge">pick</code>的过程，所以<code class="language-plaintext highlighter-rouge">otu</code>有多少，就有多少注释结果，通路上可能比第一种方法要多。</p>

<p>而且这种方法要求必须有<code class="language-plaintext highlighter-rouge">greengenes</code>的注释结果。通常我们不会用<code class="language-plaintext highlighter-rouge">greengenes</code>进行注释，所以如果肯定不会使用<code class="language-plaintext highlighter-rouge">greengenes</code>的数据库注释结果，可以采用第一种方法生成<code class="language-plaintext highlighter-rouge">biom</code>文件；</p>
<h4 id="422-对物种注释信息标准化">4.2.2 对物种注释信息标准化</h4>
<p>利用预先计算好好的<code class="language-plaintext highlighter-rouge">16s</code>拷贝数数据库，对<code class="language-plaintext highlighter-rouge">otu</code>的物种注释信息进行<code class="language-plaintext highlighter-rouge">16s</code>拷贝数的标准化。</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">normalize_by_copy_number.py <span class="nt">-i</span> otu_table.biom <span class="nt">-o</span> normalized_otus.biom
predict_metagenomes.py <span class="nt">-t</span> ko <span class="nt">-i</span> normalized_otus.biom <span class="nt">-o</span> metagenome_predictions.biom <span class="nt">-a</span> KEGG_predict_traits.tab</code></pre></figure>

<h4 id="423-功能预测">4.2.3 功能预测</h4>
<p>利用<code class="language-plaintext highlighter-rouge">predict_metagenomes.py</code>命令对标准化后的<code class="language-plaintext highlighter-rouge">otu</code>丰度进行功能预测。这里包括<code class="language-plaintext highlighter-rouge">kegg</code>, <code class="language-plaintext highlighter-rouge">cog</code>, 和<code class="language-plaintext highlighter-rouge">pfam</code>3个数据库进行预测，<code class="language-plaintext highlighter-rouge">-t cog</code> 或者 <code class="language-plaintext highlighter-rouge">-t rfam</code></p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">predict_metagenomes.py <span class="nt">-t</span> ko <span class="nt">-i</span> normalized_otus.biom <span class="nt">-o</span> metagenome_predictions.biom <span class="nt">-a</span> KEGG_predict_traits.tab</code></pre></figure>

<p>其次，加入<code class="language-plaintext highlighter-rouge">-f</code>参数， <code class="language-plaintext highlighter-rouge">predict_metagenomes.py -f -t ko -i normalized_otus.biom -o metagenome_predictions.txt -a KEGG_predict_traits.tab</code>生成<code class="language-plaintext highlighter-rouge">tab</code>键隔开的文件
得到的是每个样品中<code class="language-plaintext highlighter-rouge">KO</code>酶的丰度
<img src="/assets/picbed/tools/picrust/picrust_ko_table.png" alt="" /></p>

<h4 id="424-对预测的功能进行map划分">4.2.4 对预测的功能进行map划分</h4>
<p>将预测到的值进行<code class="language-plaintext highlighter-rouge">map</code>的划分，则<code class="language-plaintext highlighter-rouge">KO</code>的信息也是很重要，因为很多<code class="language-plaintext highlighter-rouge">KO</code>是不存在在<code class="language-plaintext highlighter-rouge">map</code>中的；</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">categorize_by_function.py <span class="nt">-f</span> <span class="nt">-i</span> metagenome_predictions.biom <span class="nt">-c</span> KEGG_Pathways <span class="nt">-l</span> 3 <span class="nt">-o</span> KEGG_Pathways.L3.txt</code></pre></figure>

<h4 id="425-是否可以用每一个通路基因在各个样品之间的值来做标准化">4.2.5 是否可以用每一个通路/基因在各个样品之间的值来做标准化。</h4>
<p>即例如<code class="language-plaintext highlighter-rouge">K01361</code>在样品<code class="language-plaintext highlighter-rouge">MFC.1</code>中，标准化之后为<code class="language-plaintext highlighter-rouge">20/(20+24+21+24)</code>。测试如下：
在挑选了<code class="language-plaintext highlighter-rouge">ref</code>序列之后，在生成<code class="language-plaintext highlighter-rouge">biom</code>文件前身的时候，由输入<code class="language-plaintext highlighter-rouge">OTU_shared_final.xls</code>文件改为输入其相对丰度文件，在<code class="language-plaintext highlighter-rouge">predict_metagenome.py</code>的过程中，会出现报错：</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">predict_metagenomes.py:371: RuntimeWarning: invalid value encountered <span class="k">in </span>true_divide

    <span class="nv">result</span><span class="o">=</span>total/n</code></pre></figure>

<p>预测的结果文件能正常生成，但是和之前的结果不一致。推测，可能在前边输入OTU的相对丰度的时候导致在16S拷贝数的标准化等等过程中有四舍五入的过程，因为虽然不一致，但是最后预测的结果是整数, 并且很多原来用绝对值去预测的时候能得到很小的值，但是换成相对丰度之后就变为了0， 有四舍五入的可能性更大了。
<img src="/assets/picbed/tools/picrust/picrust_ko_table2.png" alt="" />
且最后预测到的值，每个样品的总和是不一致的。有三种可能性，第一种是标准化之后的输入文件，在分析过程中有四舍五入的嫌疑; 第二种是虽然做了标准化，但是不是每个样品的<code class="language-plaintext highlighter-rouge">OTU</code>都能全部被挑到，这里去掉了一些<code class="language-plaintext highlighter-rouge">OTU</code>, 导致了丰度的不一致，这个是肯定的; 第三种是<code class="language-plaintext highlighter-rouge">1，2</code>两种情况同时存在。暂时我的理解是如果输入绝对丰度文件，对最后的输出结果做标准化，就相当于将所有的<code class="language-plaintext highlighter-rouge">unclassified</code>去掉之后对能注释上的物种做了一次标准化, 但因为不确定中间是否有四舍五入或者其他简略过程，需要更多的测试</p>

<h4 id="426-待测试">4.2.6 待测试</h4>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1. 可以将`pick`到`ref`的`otu`的丰度再做一次标准化，然后作为输入进行预测，计算每个样品的总和
2. `Module`没有预测出来，待解决。原文中有提到`Module`的信息，最好能预测到`module`的信息
</code></pre></div></div>

        
      </section>

      <footer class="page__meta">
        
        
  


  

  <p class="page__taxonomy">
    <strong><i class="fas fa-fw fa-tags" aria-hidden="true"></i> Tags: </strong>
    <span itemprop="keywords">
    
      <a href="/tags/amplicon" class="page__taxonomy-item p-category" rel="tag">amplicon</a><span class="sep">, </span>
    
      <a href="/tags/annotation" class="page__taxonomy-item p-category" rel="tag">annotation</a><span class="sep">, </span>
    
      <a href="/tags/pathway" class="page__taxonomy-item p-category" rel="tag">pathway</a><span class="sep">, </span>
    
      <a href="/tags/prokaryotic" class="page__taxonomy-item p-category" rel="tag">prokaryotic</a>
    
    </span>
  </p>




  


  

  <p class="page__taxonomy">
    <strong><i class="fas fa-fw fa-folder-open" aria-hidden="true"></i> Categories: </strong>
    <span itemprop="keywords">
    
      <a href="/categories/amplicon" class="page__taxonomy-item p-category" rel="tag">amplicon</a><span class="sep">, </span>
    
      <a href="/categories/bioinformatics" class="page__taxonomy-item p-category" rel="tag">bioinformatics</a>
    
    </span>
  </p>


        

  <p class="page__date"><strong><i class="fas fa-fw fa-calendar-alt" aria-hidden="true"></i> Updated:</strong> <time class="dt-published" datetime="2023-01-23">January 23, 2023</time></p>

      </footer>

      <section class="page__share">
  
    <h4 class="page__share-title">Share on</h4>
  

  <a href="https://twitter.com/intent/tweet?via=jlli6t&text=Notes+of+PICRUST%20http%3A%2F%2Flocalhost%3A3080%2Fbioinformatics%2Famplicon%2FPICRUST%2F" class="btn btn--twitter" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on Twitter"><i class="fab fa-fw fa-twitter" aria-hidden="true"></i><span> Twitter</span></a>

  <a href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Flocalhost%3A3080%2Fbioinformatics%2Famplicon%2FPICRUST%2F" class="btn btn--facebook" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on Facebook"><i class="fab fa-fw fa-facebook" aria-hidden="true"></i><span> Facebook</span></a>

  <a href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A3080%2Fbioinformatics%2Famplicon%2FPICRUST%2F" class="btn btn--linkedin" onclick="window.open(this.href, 'window', 'left=20,top=20,width=500,height=500,toolbar=1,resizable=0'); return false;" title="Share on LinkedIn"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span> LinkedIn</span></a>
</section>


      
  <nav class="pagination">
    
      <a href="/database/genome/NCBI_genomes_download/" class="pagination--pager" title="NCBI genomes下载
">Previous</a>
    
    
      <a href="/bioinformatics/amplicon/Tax4Fun/" class="pagination--pager" title="Tax4Fun
">Next</a>
    
  </nav>

    </div>

    
  </article>

  
  
    <div class="page__related">
      <h2 class="page__related-title">You may also enjoy</h2>
      <div class="grid__wrapper">
        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
    
    <h2 class="archive__item-title no_toc" itemprop="headline">
      
        <a href="/statistics/alignment/Smith-Waterman_algorithm/" rel="permalink">Smith Waterman alignment DP
</a>
      
    </h2>
    

  <p class="page__meta">
    
      
      <span class="page__meta-date">
        <i class="far fa-fw fa-calendar-alt" aria-hidden="true"></i>
        
        <time datetime="2023-02-05T00:00:00+10:00">February 5, 2023</time>
      </span>
    

    <span class="page__meta-sep"></span>

    
      
      

      <span class="page__meta-readtime">
        <i class="far fa-fw fa-clock" aria-hidden="true"></i>
        
          less than 1 minute read
        
      </span>
    
  </p>


    <p class="archive__item-excerpt" itemprop="description">The score matrix



The optimal path

</p>
  </article>
</div>

        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
    
    <h2 class="archive__item-title no_toc" itemprop="headline">
      
        <a href="/statistics/alignment/Needleman_Wunsch_alignment/" rel="permalink">Needleman Wunsch alignment DP
</a>
      
    </h2>
    

  <p class="page__meta">
    
      
      <span class="page__meta-date">
        <i class="far fa-fw fa-calendar-alt" aria-hidden="true"></i>
        
        <time datetime="2023-02-04T00:00:00+10:00">February 4, 2023</time>
      </span>
    

    <span class="page__meta-sep"></span>

    
      
      

      <span class="page__meta-readtime">
        <i class="far fa-fw fa-clock" aria-hidden="true"></i>
        
          less than 1 minute read
        
      </span>
    
  </p>


    <p class="archive__item-excerpt" itemprop="description">The score matrix



The optimal path

</p>
  </article>
</div>

        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
    
    <h2 class="archive__item-title no_toc" itemprop="headline">
      
        <a href="/database/gene/pmoA_database/" rel="permalink">pmoA基因序列数据库
</a>
      
    </h2>
    

  <p class="page__meta">
    
      
      <span class="page__meta-date">
        <i class="far fa-fw fa-calendar-alt" aria-hidden="true"></i>
        
        <time datetime="2023-02-02T00:00:00+10:00">February 2, 2023</time>
      </span>
    

    <span class="page__meta-sep"></span>

    
      
      

      <span class="page__meta-readtime">
        <i class="far fa-fw fa-clock" aria-hidden="true"></i>
        
          less than 1 minute read
        
      </span>
    
  </p>


    <p class="archive__item-excerpt" itemprop="description">Introduction
想找pmoA的序列构建数据库，找到了这个网站. 	里边包括核酸序列，以及序列对应的tax的信息。

</p>
  </article>
</div>

        
          



<div class="grid__item">
  <article class="archive__item" itemscope itemtype="https://schema.org/CreativeWork">
    
    <h2 class="archive__item-title no_toc" itemprop="headline">
      
        <a href="/linux/bash/sratoolkit/" rel="permalink">sratoolkit下载测序数据
</a>
      
    </h2>
    

  <p class="page__meta">
    
      
      <span class="page__meta-date">
        <i class="far fa-fw fa-calendar-alt" aria-hidden="true"></i>
        
        <time datetime="2021-10-29T00:00:00+10:00">October 29, 2021</time>
      </span>
    

    <span class="page__meta-sep"></span>

    
      
      

      <span class="page__meta-readtime">
        <i class="far fa-fw fa-clock" aria-hidden="true"></i>
        
          less than 1 minute read
        
      </span>
    
  </p>


    <p class="archive__item-excerpt" itemprop="description">说明
sratoolkit默认配置的下载数据目录是在HOME，这显然是不友好的，因为一般的集群给每个account的HOME目录下的存储是非常有限的，例如50Gb，所以最好把cache的目录设置到拥有大量存储可供写入的目录。
</p>
  </article>
</div>

        
      </div>
    </div>
  
  
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form class="search-content__form" onkeydown="return event.key != 'Enter';" role="search">
    <label class="sr-only" for="search">
      Enter your search term...
    </label>
    <input type="search" id="search" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
  </form>
  <div id="results" class="results"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
    <center>
      <li><strong>Follow:</strong></li>
    

    
      
        
        <li text-align:center><a href="https://scholar.google.com/citations?hl=zh-CN&user=s_Uga6sAAAAJ" title="Google scholar"><i class="fab fa-google" aria-hidden="true"></i></a></li>
        
      
        
        <li text-align:center><a href="https://github.com/jlli6t" title="Github"><i class="fab fa-fw fa-github" aria-hidden="true"></i></a></li>
        
      
        
        <li text-align:center><a href="https://www.linkedin.com/in/jie-li-3201a57b/" title="Linkedin"><i class="fab fa-linkedin" aria-hidden="true"></i></a></li>
        
      
        
        <li text-align:center><a href="https://orcid.org/0000-0002-9623-9213" title="ORCID"><i class="fa-brands fa-orcid" aria-hidden="true"></i></a></li>
        
      
        
        <li text-align:center><a href="https://twitter.com/jlli6t" title="Twitter"><i class="fab fa-fw fa-twitter-square" aria-hidden="true"></i></a></li>
        
      
    </center>

    

    
  </ul>
</div>

<script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <span id="busuanzi_container_site_pv">
<script>
  function time() {
    var date = new Date();
    var year = date.getFullYear();
    var arr_month = new Array("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")
    var month = arr_month[date.getMonth()];
    var day = date.getDate();
    var hour = date.getHours();
    var minutes = date.getMinutes();
    var seconds = date.getSeconds();
    var arr_week = new Array("Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat")
    var week = arr_week[date.getDay()]; 
    var str = day + " " + month + " " + year + " | " + week + " | " + hour + ":" + minutes + ":" + seconds;
    var div1 = document.getElementById("div1");
    div1.innerHTML = str;
    }
  setInterval(time,1000);
</script>
<center><div class="page__footer-copyright"><span id="div1"></span> <font color=F08F8F> ❤</font> the <span id="busuanzi_value_site_pv"></span>th visit </div></center>
  <center><div class="page__footer-copyright">&copy; 2023 Jie Li. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes.</a></div></center>


      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>




<script src="/assets/js/lunr/lunr.min.js"></script>
<script src="/assets/js/lunr/lunr-store.js"></script>
<script src="/assets/js/lunr/lunr-en.js"></script>







  </body>
</html>
